manager.go

```diff
@@ -2089,94 +2090,44 @@ func (m *Manager) getTracefsRegex() (*regexp.Regexp, error) {
 // Once the limit is reached, the kernel refuses to load new probes and throws a "no such device"
 // error. To prevent this, start by cleaning up the kprobe_events entries of previous managers that
 // are not running anymore.
-func (m *Manager) cleanupTracefs() error {
-	// build the pattern to look for in kprobe_events and uprobe_events
+func (m *Manager) cleanupTraceFS() error {
 	pattern, err := m.getTracefsRegex()
 	if err != nil {
-		return err
+		return fmt.Errorf("tracefs regex: %s", err)
 	}
-	// clean up kprobe_events
-	var cleanUpError error
-	pidMask := map[int]bool{Getpid(): true}
-	cleanUpError = errors.Join(cleanUpError, cleanupKprobeEvents(pattern, pidMask))
-	cleanUpError = errors.Join(cleanUpError, cleanupUprobeEvents(pattern, pidMask))
-	return cleanUpError
-}
 
-func cleanupKprobeEvents(pattern *regexp.Regexp, pidMask map[int]bool) error {
-	kprobeEvents, err := readKprobeEvents()
-	if err != nil {
-		return fmt.Errorf("couldn't read kprobe_events: %w", err)
-	}
 	var cleanUpErrors error
-	for _, match := range pattern.FindAllStringSubmatch(kprobeEvents, -1) {
-		if len(match) < 6 {
-			continue
-		}
-
-		// check if the provided pid still exists
-		pid, err := strconv.Atoi(match[5])
+	pidMask := map[int]bool{Getpid(): true}
+	eventFiles := []string{"kprobe_events", "uprobe_events"}
+	for _, eventFile := range eventFiles {
+		events, err := tracefs.ReadFile(eventFile)
 		if err != nil {
+			cleanUpErrors = errors.Join(cleanUpErrors, fmt.Errorf("read %s: %s", eventFile, err))
```

Comment:
also here, use `%w` for the error message

Reply:
See other comment about `%w`

Reply:
same comment as above